### DISCUSSION TITLE
Insight on fixing "hessian or gradient contains nan or inf value(s). Convergence halted."

### DISCUSSION THREAD

**Discussion by xgao0o** [CamDavidsonPilon/lifelines#1384] [Q&A]
I am running TimeVaryingCox on a dataset. When I fit the model, it gives convergence errors with the message: hessian or gradient contains nan or inf value(s). Any idea on where went wrong?

**Comment by CamDavidsonPilon** [comment]
Hi there, it could be a lot of things. It's possible at a specific time step (or for the full dataset), there is high colinearity between two or more columns. Have you tried adding a small l2 penalizer?
